### PR TITLE
HomeWork 17 - HomePageFactory done

### PR DESCRIPTION
**Pull request by chygraiser**

### CODE REVIEW

**Comment by chygraiser** [comment]
Hello Andrii, I have add explicit waiter /* WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("glow-ingress-line1")));*/ and commented it for myself for quick check when it will be needed. Thank You.

File: src/test/java/com/sergii/tests/factory/ProductDetailsPageFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+        log.info("Web site is opened");
+    }
+
+    @Test
+    @DisplayName(value = "Product title name test")
+    public void pdpNameTest(){
+
+        log.info("Product title name test is started");
+        String text = productDetailsPageFactory.getPdpText();
```
Comment (AndrewSkoba) [inline src/test/java/com/sergii/tests/factory/ProductDetailsPageFactoryTest.java]: No reason to use it two times

File: src/test/java/com/sergii/tests/factory/ProductDetailsPageFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        productDetailsPageFactory.getPdpText();
+        Assertions.assertThat(text).isEqualTo("TMKB 60 Percent Keyboard,Gaming Keyboard 60 Percent, LED Backlit Ultra-Compact 68 Keys Gaming Mechanical Keyboard with Separate Arrow/Control Keys, T68SE, Brown Switch");
+    }
+    @Test
+    @DisplayName(value = "Add to cart click test")
+    public void clickAddToCart(){
+
+        log.info("Add to cart click test is started");
+        productDetailsPageFactory.clickAddToCart();
+        String successMessage = driver.findElement(By.xpath("//span[@class='a-size-medium-plus a-color-base sw-atc-text a-text-bold']")).getText();
```
Comment (AndrewSkoba) [inline src/test/java/com/sergii/tests/factory/ProductDetailsPageFactoryTest.java]: Why do you search via driver here?

File: src/test/java/com/sergii/tests/factory/ProductDetailsPageFactoryTest.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    private static final Logger log = LogManager.getLogger(ProductDetailsPageFactoryTest.class);
+    private WebDriver driver;
+    private ProductDetailsPageFactory productDetailsPageFactory;
+    @BeforeEach
+    public void beforeHook() {
+
+        driver = new ChromeDriver();
+        productDetailsPageFactory = new ProductDetailsPageFactory(driver);
+        driver.get("https://www.amazon.com/TMKB-Keyboard-Ultra-Compact-Mechanical-T68SE/dp/B0BHSSZ8NJ/ref=sr_1_1_sspa");
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
```
Comment (AndrewSkoba) [inline src/test/java/com/sergii/tests/factory/ProductDetailsPageFactoryTest.java]: Try replacing it with explicit wait

File: src/test/java/com/sergii/tests/factory/ProductDetailsPageFactoryTest.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import java.time.Duration;
+
+
+public class ProductDetailsPageFactoryTest {
```
Comment (AndrewSkoba) [inline src/test/java/com/sergii/tests/factory/ProductDetailsPageFactoryTest.java:20]: Pls, solve all of the comments for HomePage tests

File: src/test/java/com/sergii/tests/factory/ProductDetailsPageFactoryTest.java (hunk 1 above)
Comment (chygraiser) [reply src/test/java/com/sergii/tests/factory/ProductDetailsPageFactoryTest.java]: done

File: src/test/java/com/sergii/tests/factory/ProductDetailsPageFactoryTest.java (hunk 2 above)
Comment (chygraiser) [reply src/test/java/com/sergii/tests/factory/ProductDetailsPageFactoryTest.java]: Done. I have create new method getAddToCartMessage for this and replace it.

File: src/test/java/com/sergii/tests/factory/ProductDetailsPageFactoryTest.java (hunk 3 above)
Comment (chygraiser) [reply src/test/java/com/sergii/tests/factory/ProductDetailsPageFactoryTest.java]: Done. I have added explicit waiters for clickAddToCartTest and clickBuyNowTest methods.

File: src/test/java/com/sergii/tests/factory/ProductDetailsPageFactoryTest.java (hunk 4 above)
Comment (chygraiser) [reply src/test/java/com/sergii/tests/factory/ProductDetailsPageFactoryTest.java:20]: Done and appropriate comments have been added. Thank you for review.

**Pull request merged**